src/etl/tissue_enrichment/tissue_enrichment.py

```diff
@@ -0,0 +1,195 @@
+"""Calculation of tissue enrichment!"""
+
+from __future__ import annotations
+
+# import numpy
+import math
+from typing import TYPE_CHECKING
+
+import pyspark.sql.functions as f
+import pyspark.sql.types as t
+from pyspark.sql.types import FloatType
+from pyspark.sql.window import Window
+from scipy import special as sps
+
+if TYPE_CHECKING:
+    from pyspark.ml.linalg import VectorUDT
+    from pyspark.sql import DataFrame
+
+
+def _melt(
+    df: DataFrame,
+    id_vars: list,
+    value_vars: list,
+    var_name: str = "variable",
+    value_name: str = "value",
+) -> DataFrame:
+    """Convert :class:`DataFrame` from wide to long format.  Copied from: https://stackoverflow.com/a/41673644.
+
+    Args:
+        df (DataFrame): input wide dataframe.
+        id_vars (list): columns names to melt to.
+        value_vars (list): values as entries for melted dataframe.
+        var_name (str): New column name for the output variables. Defaults to "variable".
+        value_name (str): New column name for the output values. Defaults to "value".
+
+    Returns:
+        DataFrame: long format dataframe
+    """
+    # Create array<struct<variable: str, value: ...>>
+    _vars_and_vals = f.array(
+        *(
+            f.struct(f.lit(c).alias(var_name), f.col(c).alias(value_name))
+            for c in value_vars
+        )
+    )
+
+    # Add to the DataFrame and explode
+    _tmp = df.withColumn("_vars_and_vals", f.explode(_vars_and_vals))
+
+    cols = id_vars + [
+        f.col("_vars_and_vals")[x].alias(x) for x in [var_name, value_name]
+    ]
+    return _tmp.select(*cols)
+
+
+def _ndtr(x: float, mean: float, std: float) -> float:
```

Comment:
I'm sure it is all fine, and there's a reason why this is the case, but when this function is called in line 93: `_ndtr(x, z, y)`, the type of these variables are `x: VectorUDT, y: VectorUDT, z: VectorUDT`. Is it fine to change the assumed type?

Reply:
Seems to have worked fine when I tried it, but will do more testing to make sure, and change back as required.